go/worker/compute/merge/committee/node.go

```diff
@@ -733,7 +734,8 @@ func (n *Node) worker() {
 				defer n.commonNode.CrossNode.Unlock()
 
 				state, ok := n.state.(StateWaitingForResults)
-				if !ok {
+				if !ok || state.timer.C != timerCh {
+					n.logger.Warn("state snapshot changed unexpectedly, aborting")
```

Comment:
No need to warn here, maybe log at debug level or just remove as this is expected to happen.